src/errors/test-run/index.js

```diff
@@ -45,14 +45,24 @@ export class MissingAwaitError extends TestRunErrorBase {
 
 // Client function errors
 //--------------------------------------------------------------------
-export class ClientFunctionExecutionInterruptionError extends TestRunErrorBase {
+export class ClientFunctionInterruptedByPageUnloadError extends TestRunErrorBase {
     constructor (instantiationCallsiteName) {
-        super(TYPE.clientFunctionExecutionInterruptionError);
+        super(TYPE.clientFunctionInterruptedByPageUnloadError);
 
         this.instantiationCallsiteName = instantiationCallsiteName;
     }
 }
 
+export class ClientFunctionInterruptedByDialogError extends TestRunErrorBase {
+    constructor (instantiationCallsiteName, dialogType, dialogText) {
+        super(TYPE.clientFunctionInterruptedByDialogError);
+
+        this.instantiationCallsiteName = instantiationCallsiteName;
+        this.dialogType                = dialogType;
+        this.dialogText                = dialogText;
```

Comment:
I think we can go without `dialogText` here. It may be huge and break error reporting